drivers/gpu/drm/vc4/vc4_dsi.c

```diff
@@ -1717,6 +1719,15 @@ static int vc4_dsi_bind(struct device *dev, struct device *master, void *data)
 	pm_runtime_enable(dev);
 
 	return 0;
+
+err_free_dma:
+	if (dsi->reg_dma_chan)
```

Comment:
Allocating a channel is dependent on ```if (dsi->variant->broken_axi_workaround)``` which is only set for a compatible string of ```bcm2835-dsi1``` (not ```bcm2835-dsi0``` or ```bcm2711-dsi1```). So it is totally possible to get here without a channel due to some other failure when using dsi0 or 2711.

I'm happy to add in zeroing the pointers, but I don't believe there is a path where that would cause issues.
- If bind failed, you won't get an unbind. 
- If already called unbind, you won't get another unbind, and they will have been initialised again before this error path can be taken again in a new bind.

Reply:
I wasn't suggesting there was a real bug here, and you've accounted for one of the tests, so I'm OK as it is.

Reply:
I'd already updated it to zero the pointers, so it is done.